docstore/gcpfirestore/query.go

```diff
@@ -139,7 +138,7 @@ func evaluateFilter(f driver.Filter, doc driver.Document) bool {
 	// Compare times.
 	if t1, ok := val.(time.Time); ok {
 		if t2, ok := f.Value.(time.Time); ok {
-			return applyComparison(f.Op, compareTimes(t1, t2))
+			return applyComparison(f.Op, driver.CompareTimes(t1, t2))
```

Comment:
It might be an edge case, but I wonder whether the behavior is consistent with the backend service, I mean, would the result be different when local filters and query filters got swapped due to change of order of the `Where` clauses?

Reply:
I agree that is something we should test. I haven't done that. I opened #2601.

Meanwhile, this PR preserves existing behavior; it's just a refactor.